master/internal/scheduler/kubernetes_resource_provider.go

```diff
@@ -272,7 +297,12 @@ func (k *kubernetesResourceProvider) receivePodTerminated(
 	container.mustTransition(containerTerminated)
 	container.exitStatus = msg.ContainerStopped
 
-	// TODO(DET-3422): de-register proxying info.
+	if names, ok := k.registeredNames[container]; ok {
```

Comment:
non-blocking: This is unnecessary, since the body will just be a no-op if the key isn't present. (And if you remove the check I suggested elsewhere, the branch will always be taken anyway.)

Reply:
yep now that we removed the empty check, we don't need this